src/main/java/com/teragrep/rlp_03/EventLoop.java

```diff
@@ -0,0 +1,134 @@
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.rlp_03;
+
+import com.teragrep.rlp_03.context.ConnectionContext;
+import com.teragrep.rlp_03.context.Context;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.channels.CancelledKeyException;
+import java.nio.channels.SelectionKey;
+import java.nio.channels.Selector;
+import java.util.Set;
+
+public class EventLoop implements AutoCloseable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventLoop.class);
+
+    private final Selector selector;
+
+    public EventLoop(Selector selector) {
+        this.selector = selector;
+    }
+
+    public Selector selector() {
+        return selector;
+    }
+
+    public void poll() throws IOException {
+        int readyKeys = selector.select();
+
+        LOGGER.debug("readyKeys <{}>", readyKeys);
+
+        Set<SelectionKey> selectionKeys = selector.selectedKeys();
+        LOGGER.debug("selectionKeys <{}> ", selectionKeys);
+        for (SelectionKey selectionKey : selectionKeys) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER
+                        .debug(
+                                "selectionKey <{}>: " + "isValid <{}>, " + "isConnectable <{}>, "
+                                        + "isAcceptable <{}>, " + "isReadable <{}>, " + "isWritable <{}>",
+                                selectionKey, selectionKey.isValid(), selectionKey.isConnectable(),
+                                selectionKey.isAcceptable(), selectionKey.isReadable(), selectionKey.isWritable()
+                        );
+            }
+
+            if (selectionKey.isAcceptable()) {
+                // ListenContext
+                ListenContext listenContext = (ListenContext) selectionKey.attachment();
+                listenContext.handleEvent(selectionKey);
+            }
+            else if (selectionKey.isConnectable()) {
+                if (selectionKey.isConnectable()) {
```

Comment:
Doubled same if?

Reply:
removed now, was left due to refactoring